hledger-lib/Hledger/Reports/AccountTransactionsReport.hs

```diff
@@ -94,63 +95,71 @@ triCommodityBalance c = filterMixedAmountByCommodity c  . triBalance
   where
-    -- a depth limit should not affect the account transactions report
-    -- seems unnecessary for some reason XXX
-    reportq = simplifyQuery $ And [_rsQuery rspec, periodq, excludeforecastq (forecast_ ropts)]
+    -- a depth limit should not affect the account transactions report, nor
+    -- should cur or amt queries, as they are handled separately
+    reportq = simplifyQuery $ And [depthlessq, periodq, excludeforecastq (forecast_ ropts)]
       where
+        depthlessq = filterQuery (not . queryIsSymOrAmt) . filterQuery (not . queryIsDepth) $ _rsQuery rspec
         periodq = Date . periodAsDateSpan $ period_ ropts
         -- Except in forecast mode, exclude future/forecast transactions.
         excludeforecastq (Just _) = Any
         excludeforecastq Nothing  =  -- not:date:tomorrow- not:tag:generated-transaction
           And [ Not . Date $ DateSpan (Just . addDays 1 $ _rsDay rspec) Nothing
               , Not generatedTransactionTag
               ]
-    symq    = filterQuery queryIsSym reportq
-    realq   = filterQuery queryIsReal reportq
-    statusq = filterQuery queryIsStatus reportq
+    symq = filterQuery queryIsSymOrAmt $ _rsQuery rspec
+    queryIsSymOrAmt q = queryIsSym q || queryIsAmt q
 
-    -- sort by the transaction's register date, for accurate starting balance
-    -- these are not yet filtered by tdate, we want to search them all for priorps
-    transactions =
-        ptraceAtWith 5 (("ts4:\n"++).pshowTransactions)
-      . sortOn (transactionRegisterDate reportq thisacctq)
-      . jtxns
-      . ptraceAtWith 5 (("ts3:\n"++).pshowTransactions.jtxns)
-      -- maybe convert these transactions to cost or value
-      . journalApplyValuationFromOpts rspec
-      . (if filtertxns then filterJournalTransactions reportq else id)
-      -- keep just the transactions affecting this account (via possibly realness or status-filtered postings)
-      . traceAt 3 ("thisacctq: "++show thisacctq)
-      . ptraceAtWith 5 (("ts2:\n"++).pshowTransactions.jtxns)
-      . filterJournalTransactions thisacctq
-      . filterJournalPostings (And [realq, statusq])
-      . ptraceAtWith 5 (("ts1:\n"++).pshowTransactions.jtxns)
-      -- apply any cur:SYM filters in reportq
-      $ if queryIsNull symq then j else filterJournalAmounts symq j
+    -- Note that within this functions, we are only allowed limited
+    -- transformation of the transaction postings: this is due to the need to
+    -- pass the original transactions into accountTransactionsReportItem.
+    -- Generally, we either include a transaction in full, or not at all.
+    -- We only allow ourselves to:
+    --   - filter amounts (e.g. cur: or amt:)
+    --   - apply valuation or costing
+    --
+    -- accountTransactionsReportItem will keep transactions of any date which
+    -- have any posting inside the report period.
+    -- Should we also require that transaction date is inside the report period ?
```

Comment:
This comment about dates might be obsolete, related to --txn-dates, and it's unclear - perhaps we should just drop it ?

Reply:
I agree. I couldn't find any code which specifically implements this, so I think it's no longer accurate.